### ISSUE TITLE
Selection Method

### ISSUE THREAD

**Issue by trav-d13** [kissinger99/MYGA#4]
1. how can I send you a question on GitHub without making any pull request? Isn't there a simple chat or how can I leave a note on the new code I have updated with the selection file so that I don't disturb you at this hour?

2. why in Selection.py, line 20 we have to adjust the current prob value? By adding back the next probability value, it will obviously be higher. Doesn't this create a misrepresentation of the actual probability of the fitness value of an individual?

**Comment by trav-d13** [comment]
1. You can create an issue in GitHub on this page, and tag me in the Assignees section, I will receive a notification and we can hold a discussion here. 

2. Are you referring to the below code:
``` 
def construct_roulette_wheel(population):
    current_prob = 0  # Initialize the current porb value
    wheel = dict()  # Create a dict to hold the probability intervals and link to the correct individuals
    for individual in population.individuals:  # Loop through each individual in the population
        probability = individual.fitness / population.calc_total_fitness()  # Calculate the inidivudals prob of selection
        current_prob = current_prob + probability  # Adjust the current prob value
        wheel[current_prob] = individual  # Assign the prob interval to the individual
    return wheel
 ```
 
 Where current probability is updated. 
 What we are doing here is we take an individual's current probability and normalize it, so that adding all of the individual's normalized probabilities adds to 1. We then construct the roulette wheel by starting from 0 and finding out to where the probability extends to say [0, 0.05] then we know if the probability lies in that range, then the individual is selected. 
 It is updated based on the next individual such that the range becomes [0.05, 0.09] and so on. The result is that fitter individuals will have a bigger range of probability and are more likely to be selected. 
 Let me know if that helped, I am happy to explain further.

**Comment by kissinger99** [comment]
Right! It makes absolutely sense. Thank you for clarifying Travis.

**Comment by kissinger99** [comment]
I have finally finalized the Selection methods: Roulette Wheel and Tournament. I've also started checking and downloading data sets for the other project. Aiming to finalize it between tomorrow and Saturday.

**Comment by trav-d13** [comment]
Fantastic to hear!!! 
Looking forward to our next session!
